src/test/java/net/minecraftforge/debug/NetworkCapabilityTest.java

```diff
@@ -0,0 +1,173 @@
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug;
+
+import java.util.function.Supplier;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.core.Direction;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.CapabilityToken;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.capabilities.INetworkCapability;
+import net.minecraftforge.common.capabilities.RegisterCapabilitiesEvent;
+import net.minecraftforge.common.util.LazyOptional;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.event.entity.living.LivingEntityUseItemEvent;
+import net.minecraftforge.event.entity.player.PlayerInteractEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.network.ForgeNetwork;
+
+@Mod(NetworkCapabilityTest.ID)
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.FORGE, modid = NetworkCapabilityTest.ID)
+public class NetworkCapabilityTest
+{
+    public static final String ID = "network_capability_test";
+
+    private static final Logger logger = LogManager.getLogger();
+
+    private static final Capability<TestCapability> TEST_CAP = CapabilityManager.get(new CapabilityToken<>() {
+    });
+
+    public NetworkCapabilityTest()
+    {
+        FMLJavaModLoadingContext.get().getModEventBus().addListener(NetworkCapabilityTest::handleRegisterCapabilities);
+    }
+
+    private static void handleRegisterCapabilities(RegisterCapabilitiesEvent event)
+    {
+        event.register(TestCapability.class);
+    }
+
+    private static class TestCapability {
+
+        private int number;
+        private boolean dirty;
+
+        public void update()
+        {
+            this.number++;
+            this.dirty = true;
+        }
+    }
+
+    private static class TestCapabilityProvider implements ICapabilityProvider, INetworkCapability {
```

Comment:
Should be serializable to test initial sync.

Reply:
I assume you mean `INBTSerializable`?

Reply:
That, or you could use `ICapabilitySerializable` over `ICapabilityProvider`